internal/service/alarms/internal/serviceconfig/serviceconfig.go

```diff
@@ -0,0 +1,300 @@
+func generateCronJob(ctx context.Context, pg *pgxpool.Pool, hubclient client.Client, configMapName string) (batchv1.CronJob, error) {
+	pgimage, err := getPostgresImage(ctx, hubclient)
+	if err != nil {
+		return batchv1.CronJob{}, fmt.Errorf("get postgres image: %w", err)
+	}
+
+	pgEnv := []corev1.EnvVar{
+		{Name: "PGHOST", Value: pg.Config().ConnConfig.Host},
+		{Name: "PGPORT", Value: strconv.Itoa(int(pg.Config().ConnConfig.Port))},
+		{Name: "PGDATABASE", Value: pg.Config().ConnConfig.Database},
+		{Name: "PGUSER", Value: pg.Config().ConnConfig.User},
+		{
+			Name: "PGPASSWORD",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: "postgres-server-passwords",
+					},
+					Key: utils.AlarmsPasswordEnvName,
+				},
+			},
+		},
+		{Name: "PGAPPNAME", Value: cleanupCronJobName},
+		{Name: "PGCONNECT_TIMEOUT", Value: "10"},
+	}
+
+	return batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cleanupCronJobName,
+			Namespace: utils.DefaultNamespace,
+			Labels:    commonLabels,
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule:                   cronSchedule,
+			ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+			SuccessfulJobsHistoryLimit: &successLimit,
+			FailedJobsHistoryLimit:     &failureLimit,
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					BackoffLimit: &jobBackoffLimit,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							RestartPolicy: corev1.RestartPolicyOnFailure,
+							Containers: []corev1.Container{
+								{
+									Name:    cleanupCronJobName,
+									Image:   pgimage,
+									Command: []string{"psql"},
+									Args:    []string{"-f", fmt.Sprintf("%s/%s", cleanScriptDir, cleanScriptName)},
+									Env:     pgEnv,
+									Resources: corev1.ResourceRequirements{
+										Limits: corev1.ResourceList{
+											corev1.ResourceCPU:    resource.MustParse("100m"),
+											corev1.ResourceMemory: resource.MustParse("128Mi"),
+										},
+									},
+									VolumeMounts: []corev1.VolumeMount{
+										{
+											Name:      configMapName,
+											MountPath: cleanScriptDir,
+										},
+									},
+								},
+							},
+							Volumes: []corev1.Volume{{
+								Name: configMapName,
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: configMapName,
+										},
+									},
+								},
+							}},
+						},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+// getPostgresImage dynamically collect the existing PG image to run psql from
+func getPostgresImage(ctx context.Context, c client.Client) (string, error) {
+	var deploymentList appsv1.DeploymentList
+	if err := c.List(ctx, &deploymentList, client.InNamespace(utils.DefaultNamespace),
+		client.MatchingLabels{"app": "postgres-server"}); err != nil {
```

Comment:
We should use our constants instead of raw "postgres-server"

Reply:
using env and removed this